build/tfs/common/publish.ts

```diff
@@ -183,30 +183,32 @@ async function publish(commit: string, quality: string, platform: string, type:
 	const blobService = azure.createBlobService(storageAccount, process.env['AZURE_STORAGE_ACCESS_KEY_2'])
 		.withFilter(new azure.ExponentialRetryPolicyFilter(20));
 
-	const mooncakeBlobService = azure.createBlobService(storageAccount, process.env['MOONCAKE_STORAGE_ACCESS_KEY'], `${storageAccount}.blob.core.chinacloudapi.cn`)
-		.withFilter(new azure.ExponentialRetryPolicyFilter(20));
-
-	// mooncake is fussy and far away, this is needed!
-	mooncakeBlobService.defaultClientRequestTimeoutInMs = 10 * 60 * 1000;
+	await assertContainer(blobService, quality);
 
-	await Promise.all([
-		assertContainer(blobService, quality),
-		assertContainer(mooncakeBlobService, quality)
-	]);
-
-	const [blobExists, moooncakeBlobExists] = await Promise.all([
-		doesAssetExist(blobService, quality, blobName),
-		doesAssetExist(mooncakeBlobService, quality, blobName)
-	]);
+	const blobExists = await doesAssetExist(blobService, quality, blobName);
 
 	const promises = [];
 
 	if (!blobExists) {
 		promises.push(uploadBlob(blobService, quality, blobName, file));
 	}
 
-	if (!moooncakeBlobExists) {
-		promises.push(uploadBlob(mooncakeBlobService, quality, blobName, file));
+	if (process.env['MOONCAKE_STORAGE_ACCESS_KEY']) {
+		const mooncakeBlobService = azure.createBlobService(storageAccount, process.env['MOONCAKE_STORAGE_ACCESS_KEY'], `${storageAccount}.blob.core.chinacloudapi.cn`)
+			.withFilter(new azure.ExponentialRetryPolicyFilter(20));
+
+		// mooncake is fussy and far away, this is needed!
+		mooncakeBlobService.defaultClientRequestTimeoutInMs = 10 * 60 * 1000;
+
+		await assertContainer(mooncakeBlobService, quality);
+
+		const mooncakeBlobExists = await doesAssetExist(mooncakeBlobService, quality, blobName);
+
+		if (!mooncakeBlobExists) {
+			promises.push(uploadBlob(mooncakeBlobService, quality, blobName, file));
+		}
+	} else {
+		console.log('Skipping Mooncake publishing.');
```

Comment:
do we need all this mooncake code?

Reply:
@kburtram Your thoughts?

Reply:
let's leave it for now until we determine what's our long term release publishing story.  Download Center has some nice features, but the process is a bit cumbersome.